week2/solution_to_week2_homework.py

```diff
@@ -0,0 +1,126 @@
+# 1.Write a function called fizz_buzz that takes a number
+# a. If the number is divisible by 3, it should return “Fizz”.
+# b. If it is divisible by 5, it should return “Buzz”.
+# c. If it is divisible by both 3 and 5, it should return “FizzBuzz”.
+# d. Otherwise, it should return the same number.
+
+def fizzbuzz(n):
+    if n % 3 == 0 and n % 5 == 0:
+        print("fizz buzz")
+    elif n % 3 == 0:
+        print("fizz")
+    elif n % 5 == 0:
+        print("buzz")
+    else:
+        print(n)
+
+number=int(input("enter a number:"))
+fizzbuzz(number)
+
+#2.Consider a lst= [5, 10, 20] and write a try and except block to avoid IndexError.
+
+lst= [5, 10, 20]
+
+def find_index(n):
+    try:
+        print(lst[n])
+    except IndexError:
+        print("index not found")
+
+
+find_index(4)
+
+#3.Create a class of Jet inventory with two arguments i.e name and country.
+# Also add the minimum 2 items in the class and print them.
+
+class Jet:
+    def __init__(self, name, country):
+        self.name = name
+        self.country = country
+
+    def jet_Info(self):
+        print(f'{self.name} is a {self.country} airline')
+
+
+myjet = Jet("SAS", "Scandinavian")
+myjet.jet_Info()
+
+#5.Write a Python script to check whether a given key already exists in a dictionary.
+# Dictionary Initialisation
+My_Dict = {'1':'Joy', '2':'John', '3':'Joye', '4':'Jiya'}
+
+My_key = input("Enter the key(integer) to be searched: ")
+
+try:
+    My_Dict[My_key]
+except KeyError:
+    print("Key doesn't exist!")
+else:
+    print("Key exists!")
```

Comment:
this solves the problem. But here you should rather actively ask if a key exists inside a dictionary.